templates/search/search_clients_last_seen_v1/query.sql

```diff
@@ -0,0 +1,163 @@
+WITH
+  _derived_search_cols AS (
+    SELECT
+      CASE
+          WHEN engine IS NULL THEN 'none'
+          WHEN STARTS_WITH(engine, 'google') THEN 'google'
+          WHEN STARTS_WITH(engine, 'ddg') OR STARTS_WITH(engine, 'duckduckgo') THEN 'ddg'
+          WHEN STARTS_WITH(engine, 'bing') THEN 'bing'
+          WHEN STARTS_WITH(engine, 'yandex') THEN 'yandex'
+          ELSE 'other'
+      END AS short_engine,
```

Comment:
Is this the proper breakdown for engine? So for e.g. google, do we want to coalesce all google searches into the same bucket, or should some google engines be treated differently than others? (if we know a certain google engine doesn't provide revenue, for example, we should perhaps exclude it here.)

Reply:
I'd opt for using the standard `engine` field without bucketing for the top-level dataset; we can always bucket accordingly for specific analyses. For example, we have a few different google codes, and this in theory can change, as could our list of partners. 

I see this dataset being useful outside of the scope of LTV as well.

Reply:
Alternatively, we could handle STARTS_WITH(engine, 'other-') as 'other' and use the standard engine name otherwise, which would reduce the cardinality. These engines never generate revenue as well, so we could opt to simply exclude `other-*` engines and keep the rest.

Reply:
I'm not totally sure this dataset will be useful outside the scope of LTV:
- The search counts are the total for each month over the past year, and this is actual month (so e.g. sept, oct) which means the values are covering different #s of days
- We are currently planning on running on a sample, so these will be inaccurate to some degree